cabal.project

```diff
@@ -55,26 +55,31 @@ source-repository-package
     type: git
     location: https://github.com/kadena-io/pact.git
     tag: 842fbc4256b3cbbde337dbeaa393b649a26f1574
+    --sha256: sha256-jD4Y2NOSYT0EQHs5BRSVvu8asfvRwt5XLDuZwbcG4mk=
```

Comment:
The command to determine the sha locally is:
```
nix-prefetch-git --url https://github.com/kadena-io/pact.git --rev <SHA>
```

Reply:
Could you add that as a comment to the cabal.project file? So when people update that file they find that info?

Reply:
Could we include an action in the CI workflow that runs that command to double checks that the nix `--sha` actually matches the source `tag`?

(You may take a look at cabal CI workflows;  there are examples for actions that check that the git tree is still clean after performing some action or test.)

Reply:
@larskuhtz Earlier today, I've added [a step](https://github.com/kadena-io/chainweb-node/pull/1570/commits/1a00b59cb7fed72acf277b0a3833a92aac5ad7eb) to check the hashes during the github action build, but then later I realized that the check was redundant, because it doesn't seem to be possible to start the build with a wrong `sha256` anyway. I've tried multiple ways to fool it, like changing the source-repository-package commit hash without updating the `sha256`, but the nix build step always failed quickly, telling the correct `sha256`. As an example, check out this run: https://github.com/kadena-io/chainweb-node/actions/runs/4831581150/jobs/8609217888

> In the past (with the old nix infra) we ran into issues where nix pins didn't exactly match the sources of the respective Hackage version and there was no easy way to detected that.

I can imagine how the old nix infra running on Zeus could perform misleading builds, picking the old source folder from the nix store due to the outdated hash, but I don't think that kind of inconsistency is possible with this new setup.

That's why I've [removed](https://github.com/kadena-io/chainweb-node/pull/1570/commits/043c25a1518d992102742af4d883824aa73e9b4e) the separate `cabal.project` checks.

Reply:
Admittedly, the error message you get from the Nix build is not ideal because it doesn't tell you what exactly you need to do. So if somebody encounters this message out of the blue, it could be frustrating to try to debug it.

Reply:
We could add a hint about how to read this error to the comment that describes how to compute the sha.